File: server/internal/entity/data/product.go
package data

import "miadok-chaladok/internal/entity"

type Product struct {
	entity.GormModel
	CategoryID      uint             `json:"categoryId"`
	TasteID         uint             `json:"tasteId"`
	Name            string           `json:"name"`
	ImagePath       string           `json:"imagePath"`
	Expiration      string           `json:"expiration"`
	Category        Category         `json:"category"`
	Taste           Taste            `json:"taste"`
	Options         []Option         `json:"options"`
	Media           []Media          `json:"media"`
	ShipmentMethods []ShipmentMethod `json:"shipmentMethods" gorm:"many2many:products_shipment_methods;"`
	Reviews         []Review         `json:"reviews"`
}

File: client/src/entities/database/Product.ts
import { Category, Taste, Option, Media, ShipmentMethod, Review } from './'

export type Product = {
	id: number
	categoryId: number
	tasteId: number
    name: string
    imagePath: string
    expiration: string

    category: Category
    taste: Taste
    options: Option[]
    media: Media[]
    shipmentMethods: ShipmentMethod[]
    reviews: Review[]
}